src/main/java/org/elasticsearch/index/query/ScriptQueryParser.java

```diff
@@ -80,7 +78,9 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                 // skip
             } else if (token == XContentParser.Token.START_OBJECT) {
-                if ("params".equals(currentFieldName)) {
+                if (ScriptField.SCRIPT.match(currentFieldName)) {
+                    script = Script.parse(parser);
+                } else if ("params".equals(currentFieldName)) { // TODO remove in 2.0 (here to support old script APIs)
                     params = parser.map();
                 } else {
```

Comment:
Should we throw an error if old-style params are passed alongside new style script{..}?